yarn-project/p2p/package.json

```diff
@@ -60,6 +60,7 @@
     "@aztec/circuits.js": "workspace:^",
     "@aztec/foundation": "workspace:^",
     "@aztec/kv-store": "workspace:^",
+    "@aztec/simulator": "workspace:^",
```

Comment:
What does it need the simulator for?

Reply:
`WorldStateDB` is required for the double spend validator

Reply:
Hmm, no it needs a `NullifierSource` which can be built in a much more lightweight fashion.

`NullifierSource` defines a single function `getNullifierIndex` which can be implemented by simply calling `findLeafIndex` on an instance of MerkleTreeOperations. No need to pull in the complete simulator here.